src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String CATEGORY_IN_FILE_SUPPLY = "supply";
+    private static final String CATEGORY_IN_FILE_BUY = "buy";
+    private static final String CATEGORY_IN_FILE_RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeInFile(getResult(readFile(fromFileName)), toFileName);
+    }
+
+    public void writeInFile(String resultInString, String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(resultInString);
+        } catch (IOException e) {
+            throw new RuntimeException("The file at path " + toFileName + " was not written" + e);
+        }
+    }
+
+    public String getResult(StringBuilder readFile) {
+        String[] readFileToStringArray = readFile.toString().split(System.lineSeparator());
+        String[] oneCategoryAndPrice;
+        StringBuilder stringBuilder = new StringBuilder();
+        int sumSupply = 0;
+        int sumBuy = 0;
+        for (String oneLine : readFileToStringArray) {
+            oneCategoryAndPrice = oneLine.split(",");
+            if (oneCategoryAndPrice[0].equals(CATEGORY_IN_FILE_SUPPLY)) {
+                sumSupply += Integer.parseInt(oneCategoryAndPrice[1]);
+            } else if (oneCategoryAndPrice[0].equals(CATEGORY_IN_FILE_BUY)) {
+                sumBuy += Integer.parseInt(oneCategoryAndPrice[1]);
+            }
+        }
+        return stringBuilder.append(CATEGORY_IN_FILE_SUPPLY).append(",").append(sumSupply)
+                .append(System.lineSeparator())
+                .append(CATEGORY_IN_FILE_BUY)
+                .append(",").append(sumBuy)
+                .append(System.lineSeparator())
+                .append(CATEGORY_IN_FILE_RESULT)
+                .append(",").append(sumSupply - sumBuy).toString();
```

Comment:
Let's extract it to the separate `private` method. Remember from `checklist`:

**Divide program logic.** For example: “write to file” or “create report” - are a separate operations. So we should create separate methods for them.

```suggestion
        return creteResultString(...);
```